### PR TITLE
Refactor processor script

### PR DESCRIPTION
**Pull request by cmccandless**
Rewrote the bulk of process_results.py to take advantage of some stdlib functionality in Python.

### CODE REVIEW

**Comment by cmccandless** [comment]
If the changes in https://github.com/exercism/c-test-runner/pull/8/commits/e02ad0d4072ef3f5c8c6066395694d3ed7b361cd are not desired, I can revert that commit.

File: process_results.py [hunk 1]
```diff
@@ -2,71 +2,63 @@
...
+def get_last_test_name(filepath):
+    if filepath not in last_test_name:
+        last_test_name[filepath] = rgx_testnames.findall(
+            Path(filepath).read_text())[-1]
+    return last_test_name[filepath]
+
+
+def truncate(text, maxlength=500):
+    if len(text) > maxlength:
+        text = f"{text[:500]}\nOutput was truncated. Please limit to {maxlength} chars."
```
Comment (bergjohan) [inline process_results.py:22]: If we specify `maxlength` to something other than 500, the text will still be truncated to 500. Shouldn't it be truncated to `maxlength`?

Comment (cmccandless) [reply process_results.py:22]: Good catch; that's what I get for copy-pasting :sweat_smile:

**Comment by cmccandless** [comment]
> If you have time, I would love to see the work you've done in this repository in the x86-64 assembly test-runner as well, exercism/x86-64-assembly-test-runner. It uses the same test framework, and the files are basically the same.

I've created a [pull request over there](https://github.com/exercism/x86-64-assembly-test-runner/pull/10)

**Comment by bergjohan** [comment]
Thank you! If you revert the `makefile` changes, I'll be happy to merge this!

**Comment by cmccandless** [comment]
Reverted!

**Comment by bergjohan** [comment]
Thanks again, merged!

**Pull request merged**